ls1intum/Artemis
# [Feature] Add build logs on new result for Jenkins

### Checklist

### Motivation and Context


Artemis displays build logs in the "Build Output" section of the code editor each time there is a compilation error. This happens automatically with the Jira/Bamboo/Bitbucket setup. after the code is submitted. The Jenkins/Gitlab setup doesn't support that yet which means that the window had to be refreshed in order to display the build logs.

### Description

This PR fixes the issue. It's part of the [Jenkins Test Notification PR](https://github.com/ls1intum/jenkins-server-notification-plugin/pull/10) which adds build log support for test notifications. This PR parses the build logs and adds them into the result object. 

This PR also refactors a bit by:
-  extracting common build log parsing functions into the `BuildLogEntryService`
-  moving Jenkins build log parsing function to a new `JenkinsBuildLogParseUtils` class

### Steps for Testing

This PR must be tested in TS2

1. Log in to Artemis
2. Create a new programming exercise
3. Participate in the exercise
4. Write some faulty code and submit it
5. The build logs should appear

### Test Coverage



BuildLogEntryService.java 94%
AbstractContinuousIntegrationService.java 98%
JenkinsBuildLogParseUtils.java 97%
JenkinsInternalUrlService.java 100%
JenkinsService.java 88%

### Screenshots


![1](https://user-images.githubusercontent.com/11298674/119114430-72a6f500-ba26-11eb-8256-9619d21aed1c.gif)
